backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/modelvalidator/C_Order.java

```diff
@@ -97,5 +122,40 @@ public void assertPOReActivationAllowed(final I_C_Order order)
 			throw new AdempiereException(MSG_PO_REACTIVATION_VOID_NOT_ALLOWED)
 					.markAsUserValidationError();
 		}
+
+		if (orderCanBeReactivated(order))
+		{
+			//delete Invoice candidates for the order
+			deleteInvoiceCandidatesForReactivatedOrder(order);
+		}
+		else
+		{
+			throw new AdempiereException(MSG_PO_REACTIVATION_NOT_ALLOWED)
+					.markAsUserValidationError();
+		}
+	}
+
+	private boolean orderCanBeReactivated(final I_C_Order order)
+	{
+		final List<OrderId> orders = new ArrayList<>();
+		orders.add(OrderId.ofRepoIdOrNull(order.getC_Order_ID()));
+		List<I_C_Invoice> invoices = invoiceDAO
+				.getInvoicesForOrderIds(orders);
+
+		final boolean hasCompletedInvoices = invoiceDAO
+				.getInvoicesForOrderIds(orders)
+				.stream()
+				.filter(inv -> DocStatus.Completed.getCode().equals(inv.getDocStatus()))
+				.count() > 0;
+
+		return !hasCompletedInvoices;
+	}
+	private void deleteInvoiceCandidatesForReactivatedOrder(final I_C_Order order)
+	{
+		final List<I_C_OrderLine> orderLines = orderDAO.retrieveOrderLines(order);
```

Comment:
pls avoid the "N+1 problem".

To do that, pls introduce IInvoiceCandDAO.deleteAllByOrderId(OrderId orderId)

which queries by C_Invoice_Candidate.C_Order_ID and deletes those ICs.